modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/RunningStats.java

```diff
@@ -309,6 +333,12 @@ public RunningStats clone() {
         }
     }
 
+    public Set<String> getAllFieldNames() {
+        // NOTE: here we assume all maps used to store values hold the same set of keys (field names).
+        // As a result, we just need to read the set of keys from one of them.
```

Comment:
Instead of writing this comment you can `assert` that they all have the same keyset. That might be a little easier to scan for a reviewer and a little more paranoid for testing.